src/Workspaces/Core/Portable/SymbolId/SymbolKey.TupleTypeSymbolKey.cs

```diff
@@ -14,14 +14,14 @@ internal abstract partial class SymbolKey
     {
         private class TupleTypeSymbolKey : AbstractSymbolKey<TupleTypeSymbolKey>
         {
-            private readonly SymbolKey[] _types;
+            private readonly SymbolKey _underlyingType;
             private readonly string[] _names;
 
             internal TupleTypeSymbolKey(INamedTypeSymbol symbol, Visitor visitor)
             {
                 Debug.Assert(symbol.IsTupleType);
 
-                _types = symbol.TupleElementTypes.Select(t => GetOrCreate(t, visitor)).ToArray();
+                _underlyingType = GetOrCreate(symbol, visitor);
```

Comment:
You are correct. It seems I ran the IDE tests at work, then made another change back home and only ran the compiler tests then... My bad. The above code should lead to infinite regression :-S

One interesting thing that comes up is that now the IDE tests need `ValueTuple` to exist in order to refactor, whereas before they didn't. I'll dig in more tomorrow.